build-tools/packages/build-cli/src/commands/release/report.ts

```diff
@@ -100,80 +138,107 @@ export default class ReleaseReportCommand extends BaseCommand<typeof ReleaseRepo
         const flags = this.processedFlags;
         const versionData: PackageReleaseData = {};
 
-        const mode = flags.highest ? "version" : flags.mostRecent ? "date" : "interactive";
-
-        this.log(`Collecting version data for release groups...`);
-        /* eslint-disable no-await-in-loop */
-        // collect version data for each release group
-        for (const rg of context.repo.releaseGroups.keys()) {
-            const name = rg;
-            const repoVersion = context.getVersion(rg);
-
-            const data = await this.collectReleaseData(
-                context,
-                name,
-                repoVersion,
-                flags.days,
-                mode,
-            );
-            if (data !== undefined) {
-                versionData[name] = data;
+        const mode =
+            flags.all || flags.highest ? "version" : flags.mostRecent ? "date" : "interactive";
+        const filter = flags.releaseGroup ?? flags.package;
+        const shouldOutputFiles = flags.output !== undefined;
+        const outputPath = flags.output ?? process.cwd();
+
+        if (filter === undefined || isReleaseGroup(filter)) {
+            this.log(`Collecting version data for release groups...`);
+            /* eslint-disable no-await-in-loop */
```

Comment:
Can we update the below to use `await Promise.all(...)` rather than disabling the rule?

Reply:
I figured out a way to do this, but I'm not sure I like it. You can see it in this commit: https://github.com/microsoft/FluidFramework/pull/12122/commits/a296c8e2879cd6d8cf4c4f8ceceea42e66b1e6e5

Reply:
Actually it doesn't work as-is because collecting version data is interactive -- sometimes the user has to choose a version. It'll need more work to restructure, so leaving this for later.